core/src/impl/Kokkos_Utilities.hpp

```diff
@@ -29,6 +29,23 @@
 namespace Kokkos {
 namespace Impl {
 
+// same as std::integral_constant but with __host__ __device__ annotations on
+// the implicit conversion function and the call operator
+template <class T, T v>
+struct integral_constant {
+  using value_type         = T;
+  using type               = integral_constant<T, v>;
+  static constexpr T value = v;
+  KOKKOS_FUNCTION constexpr operator value_type() const noexcept {
```

Comment:
If we're re-implementing it anyway because of the need for `__host__ __device__` we could eventually deprecate this conversion if we wanted to make the view API more similar to mdspan.

That's assuming this utility class is not used elsewhere...

Reply:
You read my mind

Reply:
But not doing that for a while.  We need to get users off of View::Rank first and we want to make it reasonably easy for them to support a range of Kokkos versions.